### PR TITLE
12 - Improve and update sc documentation massa docs 2/3

### PR DESCRIPTION
**Pull request by Thykof**

### CODE REVIEW

File: web3-dev/massa-web3.rst [hunk 1]
```diff
@@ -24,3 +24,184 @@ You'll find examples of frontends using the `massa-web3` library in the
...
+  * having `Thyra <https://github.com/massalabs/thyra/>`_ installed and running on your computer
+  * having a wallet in Thyra
+  * the wallet needs to own some Massa
+  * having a smart-contract :ref:`deployed <sending-sc-sum>`
+
+**Network**: as a blockchain is a network of nodes, there could be several
+networks. The main network is called the mainnet. There is also a testnet,
+run by Massa team, where you can build application without paying real Massa
+tokens. You can also run your own network, with only one node
+(see :ref:`local-network-generation`).
```
Comment (longgamma-massa) [inline web3-dev/massa-web3.rst:41]:
```suggestion
**Network**: as any blockchain is a network of nodes, there could be practically several
networks of different nodes. The main Massa network is called the MainNet. There is also a TestNet which is 
ran by Massa Labs team, where you can build applications without paying real Massa
coins. You can also run your own network, with only one node
(see :ref:`local-network-generation`).
```

File: web3-dev/massa-web3.rst [hunk 2]
```diff
@@ -24,3 +24,184 @@ You'll find examples of frontends using the `massa-web3` library in the
...
+
+A decentralized application is an application running on a blockchain. Even the
+frontend can be hosted on the blockchain with
+:ref:`web-on-chain <web3-decentralized-web>`.
+
+To build such application you will use some libraries:
```
Comment (longgamma-massa) [inline web3-dev/massa-web3.rst:47]:
```suggestion
To build such application you will need to use some libraries:
```

File: web3-dev/massa-web3.rst [hunk 3]
```diff
@@ -24,3 +24,184 @@ You'll find examples of frontends using the `massa-web3` library in the
...
+  * `massa toolkit <https://github.com/massalabs/massa-sc-toolkit/>`_
+  * `massa assemblyscript sdk <https://github.com/massalabs/massa-as-sdk/>`_
+  * `massa javascript web3 <https://github.com/massalabs/massa-web3/>`_
+
+Then you will be able to build a frontend application that communicate with
```
Comment (longgamma-massa) [inline web3-dev/massa-web3.rst:52]:
```suggestion
Then you will be able to build a frontend application that communicates with
```

File: web3-dev/massa-web3.rst [hunk 4]
```diff
@@ -24,3 +24,184 @@ You'll find examples of frontends using the `massa-web3` library in the
...
+..
+  comment: TODO: add link to the page explaining how to deploy a website
+
+Here you can find an example dapp for the :ref:`sum smart-contract <sc-example-sum>`: 
+`Massa DApp example: Sum <https://github.com/massalabs/massa-dapp-example-sum>`_.
+
+Debugging
+#########
+
+This section will give you some tips to help you debug your dapp while building it.
```
Comment (longgamma-massa) [inline web3-dev/massa-web3.rst:66]:
```suggestion
This section will give you some tips on how to debug your dapp while building it.
```

File: web3-dev/massa-web3.rst [hunk 5]
```diff
@@ -24,3 +24,184 @@ You'll find examples of frontends using the `massa-web3` library in the
...
+
+Work in local
```
Comment (longgamma-massa) [inline web3-dev/massa-web3.rst:68]: This sounds strange. Maybe work in local environment?? I am not sure is it a tech phrase 'work in local', better check with @gregLibert

File: web3-dev/massa-web3.rst [hunk 6]
```diff
@@ -24,3 +24,184 @@ You'll find examples of frontends using the `massa-web3` library in the
...
+
+.. code-block::
+
+    WALLET_PRIVATE_KEY=""
+    JSON_RPC_URL_PUBLIC=http://127.0.0.1:33035
+    JSON_RPC_URL_PRIVATE=http://127.0.0.1:33034
+
+Use the private key that the node-client gave you when calling
+`wallet_generate_secret_key`. This wallet also needs to have some Massa
+tokens, so it must appear in `massa-node/base_config/initial_ledger.json`
```
Comment (longgamma-massa) [inline web3-dev/massa-web3.rst:105]:
```suggestion
coins, so it must appear in `massa-node/base_config/initial_ledger.json`
```

File: web3-dev/massa-web3.rst [hunk 7]
```diff
@@ -24,3 +24,184 @@ You'll find examples of frontends using the `massa-web3` library in the
...
+
+.. code-block::
+
+    ip = "127.0.0.1"
+    private_port = 33034
+    public_port = 33035
+
+Start a client with the command `cargo run` in `massa/massa-client` directory.
+
+You might want to sign transaction to call smart-contract in your frontend
```
Comment (longgamma-massa) [inline web3-dev/massa-web3.rst:119]:
```suggestion
You might want to sign transactions to call smart contracts in your frontend
```

File: web3-dev/massa-web3.rst [hunk 8]
```diff
@@ -24,3 +24,184 @@ You'll find examples of frontends using the `massa-web3` library in the
...
+      .request(options)
+      .then(function (response) {
+        return response.data;
+      })
+      .catch(function (error) {
+        console.error(error);
+      });
+
+This performs an HTTP POST call to Thyra, asking to create a transaction
+that will call the function named `hello` of the smart-contract located at
```
Comment (longgamma-massa) [inline web3-dev/massa-web3.rst:151]:
```suggestion
that will call the function named `hello` of the smart contract located at
```

File: web3-dev/massa-web3.rst [hunk 9]
```diff
@@ -24,3 +24,184 @@ You'll find examples of frontends using the `massa-web3` library in the
...
+`A1nyzu9rJKnf2zz8F7mkM5d6ZoThnMuAtUdij9WmcwXWMWJFmEB` with the wallet name
+`wallet`.
+
+This opens a password prompt and you will be able to see the response of
+the call in the development tools of the browser. The response is the operation
+id, like for example `2mLLkdKB4BY4hJQFNwGZ5oQVhky13EVZAwHJXCGQnd3FRHUoHw`.
```
Comment (longgamma-massa) [inline web3-dev/massa-web3.rst:157]:
```suggestion
id, for example `2mLLkdKB4BY4hJQFNwGZ5oQVhky13EVZAwHJXCGQnd3FRHUoHw`.
```

File: web3-dev/massa-web3.rst [hunk 10]
```diff
@@ -24,3 +24,184 @@ You'll find examples of frontends using the `massa-web3` library in the
...
+
+If the operation failed, the response contains the details.
```
Comment (longgamma-massa) [inline web3-dev/massa-web3.rst:159]:
```suggestion
If the operation failed, the response will contain the details.
```

File: web3-dev/massa-web3.rst [hunk 11]
```diff
@@ -24,3 +24,184 @@ You'll find examples of frontends using the `massa-web3` library in the
...
+
+Now you can now use the commands of the node client to see the details of your
```
Comment (longgamma-massa) [inline web3-dev/massa-web3.rst:161]:
```suggestion
Now you can now use commands of the node client to see the details of your
```

**Review by longgamma-massa — CHANGES_REQUESTED** [review]
added all language fixes and there's one small comment for Greg to check

File: web3-dev/massa-web3.rst (hunk 5 above)
Comment (Thykof) [reply web3-dev/massa-web3.rst:68]: You're right I improve the title

File: web3-dev/massa-web3.rst [hunk 12]
```diff
@@ -24,3 +24,193 @@ You'll find examples of frontends using the `massa-web3` library in the
...
+
+Building Decentralized Application
+##################################
+
+For the decentralized website part, we'll assume that you have some familiarity
+with HTML and JavaScript. If you want to have more details, you can follow
+`this great tutorial from React <https://reactjs.org/tutorial/tutorial.html>`_
+from which the dApp is inspired from.
+
+Here you can find an example dapp for the :ref:`sum smart-contract <sc-example-sum>`:
```
Comment (longgamma-massa) [inline web3-dev/massa-web3.rst:36]:
```suggestion
Here you can find an example dApp for the :ref:`sum smart-contract <sc-example-sum>`: 
```

File: web3-dev/massa-web3.rst [hunk 13]
```diff
@@ -24,3 +24,193 @@ You'll find examples of frontends using the `massa-web3` library in the
...
+    `target parameter` is empty because parameters are encoded into bytes so it is
+    unlikely to have printable characters. 
+
+You can also see the events emitted by your contract:
+
+.. code-block::
+
+    get_filtered_sc_output_event caller_address=A1nyzu9rJKnf2zz8F7mkM5d6ZoThnMuAtUdij9WmcwXWMWJFmEB
+
+(assuming that your contract is deployed at `A1nyzu9rJKnf2zz8F7mkM5d6ZoThnMuAtUdij9WmcwXWMWJFmEB`).
```
Comment (longgamma-massa) [inline web3-dev/massa-web3.rst:204]: remove it here since I added it above

File: web3-dev/massa-web3.rst [hunk 14]
```diff
@@ -24,3 +24,193 @@ You'll find examples of frontends using the `massa-web3` library in the
...
+      - target parameter:
+      - max_gas:700000000
+      - gas_price:0.000001
+      - coins:0
+
+.. note::
+    `target parameter` is empty because parameters are encoded into bytes so it is
+    unlikely to have printable characters. 
+
+You can also see the events emitted by your contract:
```
Comment (longgamma-massa) [inline web3-dev/massa-web3.rst:198]:
```suggestion
You can also see events emitted by your contract (assuming that your contract is deployed at `A1nyzu9rJKnf2zz8F7mkM5d6ZoThnMuAtUdij9WmcwXWMWJFmEB`):
:
```

**Review by longgamma-massa — CHANGES_REQUESTED** [review]

File: web3-dev/massa-web3.rst (hunk 13 above)
Comment (Thykof) [reply web3-dev/massa-web3.rst]: thanks, it's done

**Review by longgamma-massa — APPROVED** [review]

**Review by gregLibert — APPROVED** [review]

**Pull request merged**